pkg/kube/ready.go

```diff
@@ -89,13 +89,6 @@ type ReadyChecker struct {
 // IsReady will fetch the latest state of the object from the server prior to
 // performing readiness checks, and it will return any error encountered.
 func (c *ReadyChecker) IsReady(ctx context.Context, v *resource.Info) (bool, error) {
-	var (
-		// This defaults to true, otherwise we get to a point where
-		// things will always return false unless one of the objects
-		// that manages pods has been hit
-		ok  = true
-		err error
-	)
```

Comment:
This was interesting. Several of us have talked about this and it's hard to grok what `ok = true` does here.

Connected with @sudermanjr who connected with @joejulian about this. Joe worked to split ready out from wait from Helm 2 to Helm 3, and seems this was not really noticed during that work. This appears to be a carry over from helm v2.

What I would like to know is, does this PR work without this change? Is anyone available to try? If so, I'd clean this up as a separate PR so it could be reverted on it's own if needed later.

Reply:
The PR works without this change when I tested with the affected handling of ReplicaSets and ReplicationController,
but of course the code looks a bit weirder..
So fixing that in a separate PR should be ok.

Reply:
Wouldn't it be easier to club this change in with this PR rather than two separate PRs?

Echoing @bjosv's thoughts- feels weird to use those vars just for the [*corev1.ReplicationController, *extensionsv1beta1.ReplicaSet, *appsv1beta2.ReplicaSet, *appsv1.ReplicaSet] checks

@scottrigby